### PR TITLE
Add aliasing documentation

### PR DESCRIPTION
**Pull request by xliang6**
1. Add high-level overview on aliasing infrastructure;
2. Provide detailed information on five kinds of bit vectors
we maintain in the context of Java.

[skip ci]

Signed-off-by: Xiaoli Liang <xliang6@gmail.com>

### CODE REVIEW

File: doc/compiler/il/SymbolsSymrefsAliasing.md [hunk 1]
```diff
@@ -103,11 +103,20 @@ each bit specifies whether a corresponding symref is a member of the set or not.
...
+aliasing used in OMR compiler differs from the classical definition of aliasing. Specifically,
+it is also used as an indicator of side-effects, rather than to mean that multiple names are
 used to access the same memory. For example, a method symref may alias a static symref,
 signifying that a call to the method kills the value of the static.
 
+At a high level, aliasing information is created by calling 
+`TR::AliasBuilder::createAliasInfo`. The optimizer infrastructure invokes `createAliasInfo` 
+if alias sets have not been built yet or alias sets have been invalidated by a previous 
+optimization pass. An optimization pass will invalidate alias sets if it is creating a 
+symbol reference, i.e. inliner, PRE, EA, etc. Generally, alias information is not maintained
```
Comment (0xdaryl) [inline doc/compiler/il/SymbolsSymrefsAliasing.md]: Could you expand PRE and EA for clarity?  Those terms are not ubiquitous yet.

File: doc/compiler/il/SymbolsSymrefsAliasing.md [hunk 2]
```diff
@@ -103,11 +103,20 @@ each bit specifies whether a corresponding symref is a member of the set or not.
 ### Basic concept
 
 Aliasing information is associated with symrefs. However, the structural definition of
-aliasing used in Testarossa differs from the classical definition of aliasing. Specifically,
-it also is used as an indicator of side-effects, rather than to mean that multiple names are
+aliasing used in OMR compiler differs from the classical definition of aliasing. Specifically,
```
Comment (0xdaryl) [inline doc/compiler/il/SymbolsSymrefsAliasing.md]: the OMR compiler

File: doc/compiler/il/SymbolsSymrefsAliasing.md [hunk 3]
```diff
@@ -20,12 +20,12 @@ OpenJDK Assembly Exception [2].
...
 -->
 
-# Symbols, Symbol References, and Aliasing in Testarossa
+# Symbols, Symbol References, and Aliasing in OMR compiler
 
 ## Introduction
 
 This document explains and documents the concepts and implementation of symbols,
-symbol references, and aliasing in Testarossa.
+symbol references, and aliasing in OMR compiler.
```
Comment (0xdaryl) [inline doc/compiler/il/SymbolsSymrefsAliasing.md]: the OMR compiler

File: doc/compiler/il/SymbolsSymrefsAliasing.md [hunk 4]
```diff
@@ -20,12 +20,12 @@ OpenJDK Assembly Exception [2].
 SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
 -->
 
-# Symbols, Symbol References, and Aliasing in Testarossa
+# Symbols, Symbol References, and Aliasing in OMR compiler
```
Comment (0xdaryl) [inline doc/compiler/il/SymbolsSymrefsAliasing.md]: the OMR compiler

File: doc/compiler/il/SymbolsSymrefsAliasing.md [hunk 5]
```diff
@@ -132,6 +141,14 @@ associated bit vectors themselves are stored in the `TR::AliasBuilder` class. Fo
 given symref, each query is successively performed. Every time a query returns a
 positive result, the associated bit vector is "ORed" into the final alias bit vector.
 
+In particular, if `TR::SymbolReference::getUseDefAliasesBV()` is invoked on a shadow
```
Comment (0xdaryl) [inline doc/compiler/il/SymbolsSymrefsAliasing.md]: In particular, in the case of Java, ...

**Comment by xliang6** [comment]
@0xdaryl @vijaysun-omr Please review. Thanks.

**Comment by vijaysun-omr** [comment]
Looks okay to me.

**Pull request merged**